From e4361ed92e4f1c989bb7b0b4a9ed157e2053063b Mon Sep 17 00:00:00 2001
From: Jacob McSwain <jacob.a.mcswain@gmail.com>
Date: Fri, 3 Nov 2023 19:15:08 -0500
Subject: [PATCH 7/7] vtun: normalize hosts as lowercase internally

---
 cfg_file.y | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/cfg_file.y b/cfg_file.y
index 8552e0e..6afeb62 100644
--- a/cfg_file.y
+++ b/cfg_file.y
@@ -27,6 +27,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
+#include <ctype.h>
 
 #include <syslog.h>
 
@@ -105,6 +106,13 @@ statement: '\n'
 		   * MUST dup strings to be able to reread config.
 		   */
 	  	  memcpy(parse_host, &default_host, sizeof(struct vtun_host));
+
+		  unsigned char *hostStr = (unsigned char *) $1;
+		  while (*hostStr) {
+			*hostStr = tolower(*hostStr);
+			hostStr++;
+		  }
+		  
 		  parse_host->host = strdup($1);
 		  parse_host->passwd = NULL;
 
@@ -593,6 +601,11 @@ int free_host(void *d, void *u)
  */ 
 inline struct vtun_host* find_host(char *host)
 {
+	unsigned char *hostStr = (unsigned char *) host;
+	while (*hostStr) {
+		*hostStr = tolower(*hostStr);
+		hostStr++;
+	}
    return (struct vtun_host *)llist_free(&host_list, free_host, host);
 }
 
-- 
2.41.0

