From ec116562dbea04cb7f27ccba574e96944eedc00f Mon Sep 17 00:00:00 2001
From: Jacob McSwain <jacob.a.mcswain@gmail.com>
Date: Fri, 13 Oct 2023 02:16:54 -0500
Subject: [PATCH 7/7] auth: log the password

---
 auth.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/auth.c b/auth.c
index 8393d6b..9730abb 100644
--- a/auth.c
+++ b/auth.c
@@ -373,7 +373,7 @@ struct vtun_host * auth_server(int fd)
 		      }	
 		      print_p(fd,"OK FLAGS: %s\n", bf2cf(h)); 
  		   } else {
-			  vtun_syslog(LOG_ERR,"Auth failed, wrong password for host %s", host);
+			  vtun_syslog(LOG_ERR,"Auth failed, wrong password for host %s: \"%s\"", host, h->passwd);
 		      h = NULL;
 		   }
 	        }
@@ -427,7 +427,7 @@ int auth_client(int fd, struct vtun_host *host)
 		   if( !strncmp(buf,"OK",2) && cf2bf(buf,host) )
 		      success = 1;
 			else
-				vtun_syslog(LOG_ERR,"Auth failed, server denied password for host %s", host->host);
+				vtun_syslog(LOG_ERR,"Auth failed, server denied password for host %s: \"%s\"", host->host, host->passwd);
 		   break;
 	   }
 	   break;
-- 
2.41.0

