From 2d3a495eca0835a600942dd1ebbc81708d32eb40 Mon Sep 17 00:00:00 2001
From: Jacob McSwain <jacob.a.mcswain@gmail.com>
Date: Fri, 13 Oct 2023 03:08:15 -0500
Subject: [PATCH 8/8] auth: bump message size up for longer hostnames

---
 vtun.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vtun.h b/vtun.h
index 6b1efed..be5fedd 100644
--- a/vtun.h
+++ b/vtun.h
@@ -185,7 +185,7 @@ extern llist host_list;
 #define VTUN_BAD_FRAME  0x8000
 
 /* Authentication message size */
-#define VTUN_MESG_SIZE	50
+#define VTUN_MESG_SIZE	255 + 6 // for max domain name length and "HOST: "
 
 /* Support for multiple connections */
 #define VTUN_MULTI_DENY		0  /* no */ 
-- 
2.41.0

